Realm/Realm/Realm.cs

```diff
@@ -1448,29 +1430,32 @@ public void AddRealm(Realm realm)
             {
                 // We only want to have each realm once. That should be the case as AddRealm
                 // is only called in the Realm ctor, but let's check just in case.
-                Debug.Assert(!_weakRealms.Any(r =>
-                {
-                    return r.TryGetTarget(out var other) && ReferenceEquals(realm, other);
-                }), "Trying to add a duplicate Realm to the RealmState.");
+                Debug.Assert(!GetLiveRealms().Any(other => ReferenceEquals(realm, other)), "Trying to add a duplicate Realm to the RealmState.");
```

Comment:
Would it make sense to have this as a test instead of an assert?

Reply:
It's an assert because this is currently untestable behavior - there's no way currently to add the same realm twice as adding happens in the constructor. However, because there's nothing in the typesystem or the API design that enforces that, I wanted to add the extra check in case someone accidentally moves things around in the future. And it's a debug assert because the check isn't super cheap, so I didn't want us to do it in production.